src/model_signing/manifest.py

```diff
@@ -245,16 +246,156 @@ def key(self) -> ManifestKey:
+            allow_symlinks: Controls whether symbolic links are included.
+        """
+        self._hash_type = hash_type
+        self._allow_symlinks = allow_symlinks
+
+    @property
+    @override
+    def serialization_dict(self) -> dict[str, Any]:
+        return {
+            "method": self.method,
+            "hash_type": self._hash_type,
+            "allow_symlinks": self._allow_symlinks,
+        }
+
+    @classmethod
+    @override
+    def _from_dict(cls, serialization_dict: dict[str, Any]) -> Self:
+        return cls(
+            serialization_dict["hash_type"],
+            serialization_dict["allow_symlinks"],
+        )
+
+    @override
+    def new_item(self, name: str, digest: hashing.Digest) -> ManifestItem:
+        path = pathlib.PurePosixPath(name)
+        return FileManifestItem(path=path, digest=digest)
+
+
+class _ShardSerialization(SerializationType):
+    method: Final[str] = "shards"
+
+    def __init__(
+        self, hash_type: str, shard_size: int, allow_symlinks: bool = False
+    ):
+        """Records the manifest serialization type for serialization by files.
+
+        We need to record the hashing engine used and whether symlinks are
+        hashed or ignored, just like for file serialization. We also need to
+        record the shard size used to split the files, since different shard
+        sizes results in different resources.
+
+        Args:
+            hash_type: A string representation of the hash algorithm.
+            allow_symlinks: Controls whether symbolic links are included.
+        """
+        self._hash_type = hash_type
+        self._allow_symlinks = allow_symlinks
+        self._shard_size = shard_size
+
+    @property
+    @override
+    def serialization_dict(self) -> dict[str, Any]:
+        return {
+            "method": self.method,
+            "hash_type": self._hash_type,
+            "shard_size": self._shard_size,
+            "allow_symlinks": self._allow_symlinks,
+        }
+
+    @classmethod
+    @override
+    def _from_dict(cls, serialization_dict: dict[str, Any]) -> Self:
+        return cls(
+            serialization_dict["hash_type"],
+            serialization_dict["shard_size"],
+            serialization_dict["allow_symlinks"],
+        )
+
+    @override
+    def new_item(self, name: str, digest: hashing.Digest) -> ManifestItem:
+        parts = name.split(":")
+        if len(parts) != 3:
+            raise ValueError(
+                "Invalid resource name: expected 3 components separated by "
+                f"`:`, got {name}"
+            )
+
+        path = pathlib.PurePosixPath(parts[0])
+        start = int(parts[1])
+        end = int(parts[2])
+        return ShardedFileManifestItem(
+            path=path, start=start, end=end, digest=digest
+        )
+
+
 class Manifest:
     """Generic manifest file to represent a model."""
 
-    def __init__(self, items: Iterable[ManifestItem]):
+    def __init__(
+        self,
+        model: str,
```

Comment:
i know we usually use pathlib.Path or similar for paths, but at first glance i would wonder if this was a path or not. What do you think about `name`, `model_name` or something like that? I know the docstring below explains it, but just what came to mind.

Reply:
Right, `model_name` sounds much better